request_helpers.go

```diff
@@ -50,15 +55,40 @@ func (client *Config) url(endpoint string) string {
 	return fmt.Sprintf("%v/%v/%v", client.APIURL, client.APIVersion, endpoint)
 }
 
+func getUserAgent() string {
+	platform := runtime.GOOS
+	uname, err := exec.LookPath("uname")
+	if err == nil {
+		cmd := exec.Command(uname, "-a")
+		var out bytes.Buffer
+		cmd.Stderr = nil
+		cmd.Stdout = &out
+		err = cmd.Run()
+		if err == nil {
+			platform = strings.TrimSpace(out.String())
+		}
+	}
+
+	goVersion := runtime.Version()
+	re, _ := regexp.Compile(`go(.+)`)
+	matches := re.FindStringSubmatch(goVersion)
+	if len(matches) > 1 {
```

Comment:
Should we get the highest Go version? This looks like it just grabs the first, it might make more sense to ensure we order this consistently.

Reply:
@codymullins This is really just about formatting the string. `runtime.Version()` always returns one version for go in a format like `go1.12`. So the regex is pulling out just the version number (`1.12`) so we can end up with a format like `go 1.12`. In other words, all this is effectively doing is adding a space between `go` and `1.12`.